src/firmware.cpp

```diff
@@ -60,118 +42,104 @@ static String pop_option(String& argument) {
   }
 }
 
-static CommandError run_help(int commandId, String argument);
+static void serialWrite(int requestID, char lineType, const String& str) {
+    if (requestID != 0) {
+        Serial.write('@');
+        Serial.print(requestID, DEC);
+        Serial.write(' ');
+    }
+
+    Serial.write(lineType);
+    Serial.write(' ');
 
-static CommandError led(int commandId, String argument) {
-  if (argument == "on") {
-    digitalWrite(LED_BUILTIN, HIGH);
-  } else if (argument == "off") {
-    digitalWrite(LED_BUILTIN, LOW);
-  } else {
-    return COMMAND_ERROR("unknown argument");
-  }
-  return OK;
+    Serial.println(str);
 }
 
-static CommandError servo(int commandId, String argument) {
-  String servoArg = pop_option(argument);
-  String widthArg = pop_option(argument);
+// The actual commands
 
-  if (argument.length() || !servoArg.length() || !widthArg.length()) {
-    return COMMAND_ERROR("servo takes exactly two arguments");
-  }
+static void readAnaloguePin(int requestID, const String& name, int pin) {
+  int reading = analogRead(pin);
```

Comment:
I feel like this temporary variable is unnecessary, might as well pass it to `String` directly

Reply:
It probably reads better to have it as seperate commands, as the intent of teh funtion is seperate from the response generation, the compiler will also create a variable for it anyway,

Reply:
This variable was leftover from the old code. I have taken @RealOrangeOne's suggestion.

Reply:
e5da051